eduNEXT/platform-plugin-superset
# Safer authentication and content rendering for the Superset XBlock

### Description

This PR makes a number of changes to make the Superset XBlock "safer" for use in courses.

1. Moves sensitive authentication fields out of the XBlock fields and into into `XBLOCK_SETTINGS`, the django setting where we usually store configuration like this.
2. Clarifies configuration fields to distinguish between Superset's `internal_service_url` (formerly `host`) and the external, public `service_url`.
3. Hides Superset content from users marked as "students" in the course.

### Testing instructions

Code review is best done commit by commit.

To test, follow the updated installation instructions to install this plugin with Tutor, and add it to your course.

1. Add a new unit, and add this XBlock via Advanced > Superset Dashboard, and configure its `dashboard_uuid`.
2. Visit the new unit as a course staff member: should display configured Superset dashboard.
3. Visit the new unit as a course student: should display a message, not the Superset dashboard.
4. Visit the Instructor Dashboard, and ensure the Superset Dashboard renders there as expected.

### Author Notes & Concerns

I've got this rendering in Studio and the LMS, but not the Instructor Dashboard:  the page spins and never returns, but doesn't show any errors in the LMS either. Not sure what I'm doing wrong. 

**Merge checklist:**
Check off if complete *or* not applicable:
